### ISSUE TITLE
OpenHAB 3 TCP Binding [TCP]

### ISSUE THREAD

**Issue by syntacrsc** [openhab/openhab-addons#10428]
<a alt="bounty" href="https://www.bountysource.com/issues/97560841"><img src="https://api.bountysource.com/badge/issue?issue_id=97560841"></a>

The TCP-Binding from OpenHAB 2 should also be available for OpenHAB 3.
For details on the existing OH2 TCP-Binding see: [OpenHAB TCP-Binding documentation](https://v2.openhab.org/addons/bindings/tcp1/)

Many users already expressed their wish for such binding for OH3 (see: [openhab community on OH3 TCP binding)](https://community.openhab.org/t/openhab-3-tcp-binding/111896/19) many of those users (including myself) weren't able to continue to use their TCP-Binding based devices and implementations (e.g. rules) after upgrading to OH3.

As an incentive for developers to provide a TCP-binding for OH3 I've posted a 50 USD bounty on [Bountysource.com](https://www.bountysource.com/issues/97560841-openhab-3-tcp-binding-tcp) and I will encourage additional users to join me.

Hopefully there's someone out there who is able to develop a TCP-binding for OH3 (if I knew how to I'd do it myself - unfortunately the only programming language I really know is SAP ABAP).

**Comment by reckoning89** [comment]
[reckoning89] Interested — starting work on this bounty.

Plan: implement per acceptance criteria, add tests/docs as required, open PR within 48h.
Payment: USDC to `0x028a964901762571022C5f2C9b66717a1c25886F` on Base if applicable.

/attempt

**Comment by lsiepel** [comment]
AI PR incoming. See account details and commits from past few days.

**Comment by digitaldan** [comment]
This looks like automated AI spam searching the internet for bounties and my guess is if paid, would use AI  to one shot a solution......

(Edit sorry @lsiepel  i read your comment as "A PR incoming", not Ai ,  as if you had a real PR for this, now i see you are seeing what i was seeing)

**Comment by digitaldan** [comment]
See https://github.com/search?q=reckoning89&type=issues

**Comment by stmr** [comment]
I can take this if still available/payable.

Planned scope: OpenHAB 3 TCP Binding.

I will keep the PR/submission focused, include tests/docs where requested, run project checks, and wait for maintainer or bounty-platform confirmation before starting work.

**Comment by digitaldan** [comment]
This is why we can't have nice things.

**Comment by digitaldan** [comment]
@lsiepel should we just close this issue?  I feel like its become a honey trap now.

**Comment by lsiepel** [comment]
Yes! Let’s close this. Sorry but I have very low confidence in a true fix for this long standing request.
